modules/hostap/src/supp_main.c

```diff
@@ -523,6 +585,455 @@ static int register_supplicant_event_socket(struct supplicant_context *ctx)
+		return -1;
+	}
+
+	if (hapd->driver->get_capa && hapd->driver->get_capa(hapd->drv_priv, &capa) == 0) {
+		struct wowlan_triggers *triggs;
+
+		iface->drv_flags           = capa.flags;
+		iface->drv_flags2          = capa.flags2;
+		iface->probe_resp_offloads = capa.probe_resp_offloads;
+		/*
+		 * Use default extended capa values from per-radio information
+		 */
+		iface->extended_capa         = capa.extended_capa;
+		iface->extended_capa_mask    = capa.extended_capa_mask;
+		iface->extended_capa_len     = capa.extended_capa_len;
+		iface->drv_max_acl_mac_addrs = capa.max_acl_mac_addrs;
+
+		/*
+		 * Override extended capa with per-interface type (AP), if
+		 * available from the driver.
+		 */
+		hostapd_get_ext_capa(iface);
+
+		triggs = wpa_get_wowlan_triggers(conf->wowlan_triggers, &capa);
+		if (triggs && hapd->driver->set_wowlan) {
+			if (hapd->driver->set_wowlan(hapd->drv_priv, triggs)) {
+				wpa_printf(MSG_ERROR, "set_wowlan failed");
+			}
+		}
+		os_free(triggs);
+	}
+
+	return 0;
+}
+
+struct hostapd_config *hostapd_config_read2(const char *fname)
+{
+	struct hostapd_config *conf;
+	const struct device *dev;
+	char ifname[IFNAMSIZ + 1] = {0};
+	int errors = 0;
+	size_t i;
+	int aCWmin = 4, aCWmax = 10;
+	/* background traffic */
+	struct hostapd_wmm_ac_params ac_bk = {aCWmin, aCWmax, 9, 0, 0};
+	/* best effort traffic */
+	struct hostapd_wmm_ac_params ac_be = {aCWmin, aCWmax - 4, 5, 0, 0};
+	/* video traffic */
+	struct hostapd_wmm_ac_params ac_vi = {aCWmin - 1, aCWmin, 3,
+					      3008 / 32, 0};
+	/* voice traffic */
+	struct hostapd_wmm_ac_params ac_vo = {aCWmin - 2, aCWmin - 1, 3,
+					      1504 / 32, 0};
+
+	dev = net_if_get_device(net_if_get_wifi_sap());
+	strncpy(ifname, dev->name, IFNAMSIZ);
+	ifname[IFNAMSIZ] = '\0';
+
+	conf = hostapd_config_defaults();
+	if (conf == NULL) {
+		return NULL;
+	}
+
+	conf->wmm_ac_params[0] = ac_be;
+	conf->wmm_ac_params[1] = ac_bk;
+	conf->wmm_ac_params[2] = ac_vi;
+	conf->wmm_ac_params[3] = ac_vo;
+
+	/* set default driver based on configuration */
+	conf->driver = wpa_drivers[0];
+	if (conf->driver == NULL) {
+		wpa_printf(MSG_ERROR, "No driver wrappers registered!");
+		hostapd_config_free(conf);
+		return NULL;
+	}
+	conf->last_bss = conf->bss[0];
+	struct hostapd_bss_config *bss;
+
+	bss                 = conf->last_bss;
+	bss->start_disabled = 1;
+	bss->max_num_sta    = 8;
+	bss->dtim_period    = 1;
+	os_strlcpy(conf->bss[0]->iface, ifname, sizeof(conf->bss[0]->iface));
+	bss->logger_stdout_level = HOSTAPD_LEVEL_INFO;
+	bss->logger_stdout       = 0xffff;
+	bss->nas_identifier      = os_strdup("ap.example.com");
+	os_memcpy(conf->country, "US ", 3);
+	conf->hw_mode        = HOSTAPD_MODE_IEEE80211G;
+	bss->wps_state       = WPS_STATE_CONFIGURED;
+	bss->eap_server      = 1;
+#ifdef CONFIG_WPS
+	bss->ap_setup_locked = 1;
+#endif
+	conf->channel        = 1;
+	conf->acs            = conf->channel == 0;
+#ifdef CONFIG_ACS
+	conf->acs_num_scans  = 1;
+#endif /* CONFIG_ACS */
```

Comment:
Don't Repeat Yourself. If the condition is obvious, I suggest to not repeat it after the `#endif`.

Reply:
Removed comment after `#endif`.